File: model-client/src/main/java/org/modelix/model/api/INodeReference.java
package org.modelix.model.api;

import org.jetbrains.annotations.Nullable;

/**
 * The relation between an INodeReference and an INode is n to 1.
 * Two INodeReferences that are not equal can resolve to the same INode.
 */
public interface INodeReference {
  @Nullable
  INode resolveNode(INodeResolveContext context);
}

File: model-client/src/main/kotlin/org/modelix/model/api/INodeReference.kt
package org.modelix.model.api

/**
 * The relation between an INodeReference and an INode is n to 1.
 * Two INodeReferences that are not equal can resolve to the same INode.
 */
interface INodeReference {
    fun resolveNode(context: INodeResolveContext?): INode?
}
